omrsig fails to build against glibc 2.21 or newer

```
omrsig.cpp: In function ‘void sv_flags_to_sa_flags(const int*, int*)’:
omrsig.cpp:616:18: error: ‘SV_ONSTACK’ was not declared in this scope
  if (*sv_flags & SV_ONSTACK) {
                  ^
omrsig.cpp:620:24: error: ‘SV_INTERRUPT’ was not declared in this scope
  if (0 == (*sv_flags & SV_INTERRUPT)) {
                        ^
omrsig.cpp:623:18: error: ‘SV_RESETHAND’ was not declared in this scope
  if (*sv_flags & SV_RESETHAND) {
                  ^
omrsig.cpp: In function ‘void sa_flags_to_sv_flags(const int*, int*)’:
omrsig.cpp:649:16: error: ‘SV_ONSTACK’ was not declared in this scope
   *sv_flags |= SV_ONSTACK;
                ^
omrsig.cpp:656:16: error: ‘SV_RESETHAND’ was not declared in this scope
   *sv_flags |= SV_RESETHAND;
                ^
omrsig.cpp:659:16: error: ‘SV_INTERRUPT’ was not declared in this scope
   *sv_flags |= SV_INTERRUPT;
                ^
omrsig.cpp: In function ‘int sigvec(int, const sigvec*, sigvec*)’:
omrsig.cpp:685:24: error: invalid use of incomplete type ‘const struct sigvec’
    act.sa_handler = vec->sv_handler;
                        ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
                                  ^
omrsig.cpp:686:27: error: invalid use of incomplete type ‘const struct sigvec’
    sv_mask_to_sa_mask(&vec->sv_mask, &act.sa_mask);
                           ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
                                  ^
omrsig.cpp:687:29: error: invalid use of incomplete type ‘const struct sigvec’
    sv_flags_to_sa_flags(&vec->sv_flags, &act.sa_flags);
                             ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
                                  ^
omrsig.cpp:701:8: error: invalid use of incomplete type ‘struct sigvec’
    ovec->sv_handler = oldact.sa_handler;
        ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
                                  ^
omrsig.cpp:702:42: error: invalid use of incomplete type ‘struct sigvec’
    sa_mask_to_sv_mask(&act.sa_mask, &ovec->sv_mask);
                                          ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
                                  ^
omrsig.cpp:703:45: error: invalid use of incomplete type ‘struct sigvec’
    sa_flags_to_sv_flags(&act.sa_flags, &ovec->sv_flags);
                                             ^
In file included from omrsig.cpp:38:0:
../include_core/omrsig.h:110:34: note: forward declaration of ‘struct sigvec’
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
```

--- a/omrsigcompat/omrsig.cpp
+++ b/omrsigcompat/omrsig.cpp
@@ -38,6 +38,12 @@
 #include "omrsig.h"
 #include "omrsig_internal.hpp"
 
+#if defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 21)
+#undef OMR_OMRSIG_HAS_SIGVEC
+#else /* defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 21) */
+#define OMR_OMRSIG_HAS_SIGVEC
+#endif /* defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 21) */
+
 extern "C" {
 
 static OMR_SigData sigData[NSIG];
@@ -48,6 +54,7 @@ static int omrsig_sigaction_internal(int signum, const struct sigaction *act, st
 static bool validSignalNum(int signum, bool nullAction);
 static bool handlerIsFunction(const struct sigaction *act);
 #if !defined(WIN32) && !defined(J9ZOS390)
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
 static void sa_mask_to_sv_mask(const sigset_t *sa_mask, int *sv_mask);
 static void sv_mask_to_sa_mask(const int *sv_mask, sigset_t *sa_mask);
 #if (defined(S390) && defined(OMR_ENV_DATA64))
@@ -57,6 +64,7 @@ static void sv_flags_to_sa_flags(const int *sv_flags, long unsigned int *sa_flag
 static void sa_flags_to_sv_flags(const int *sa_flags, int *sv_flags);
 static void sv_flags_to_sa_flags(const int *sv_flags, int *sa_flags);
 #endif /* (defined(S390) && defined(OMR_ENV_DATA64)) */
+#endif /* defined(OMR_OMRSIG_HAS_SIGVEC) */
 #endif /* !defined(WIN32) && !defined(J9ZOS390) */
 
 #if defined(WIN32)
@@ -593,6 +601,8 @@ sysv_signal(int signum, sighandler_t handler) __THROW
 	return omrsig_signal_internal(signum, handler);
 }
 
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
+
 static void
 sv_mask_to_sa_mask(const int *sv_mask, sigset_t *sa_mask)
 {
@@ -661,12 +671,17 @@ sa_flags_to_sv_flags(const int *sa_flags, int *sv_flags)
 #endif /* defined(AIXPPC) */
 }
 
+#endif /* defined(OMR_OMRSIG_HAS_SIGVEC) */
+
 #endif /* !defined(J9ZOS390) */
 #endif /* !defined(WIN32) */
 
 } /* extern "C" { */
 
 #if (!defined(J9ZOS390) && !defined(WIN32))
+
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
+
 int
 #if defined(OSX)
 sigvec(int sig, struct sigvec * vec, struct sigvec *ovec)
@@ -705,5 +720,8 @@ sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW
 	}
 	return rc;
 }
+
+#endif /* defined(OMR_OMRSIG_HAS_SIGVEC) */
+
 #endif /* (!defined(J9ZOS390) && !defined(WIN32)) */
 

